domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt

```diff
@@ -95,15 +106,36 @@ class TranslationController @Inject constructor(
     }
   }
 
+  /**
+   * Returns List data for [SupportedLanguages] available language definitions from proto asset file.
+   */
+  fun getAllAppLanguageDefinitions(): List<OppiaLanguage> {
+    val supportedLanguages = languageConfigRetriever.loadSupportedLanguages()
+    val supportedLanguageList = arrayListOf<OppiaLanguage>()
+    for (LanguageSupportDefinition in supportedLanguages.languageDefinitionsList) {
+      if (LanguageSupportDefinition.hasAppStringId()) {
+        supportedLanguageList.add(LanguageSupportDefinition.language)
+      }
+    }
+    return supportedLanguageList
+  }
+
   /**
    * Returns a data provider for a [OppiaLocale.DisplayLocale] corresponding to the user's selected
    * language for app strings (see [getAppLanguage]).
    */
   fun getAppLanguageLocale(profileId: ProfileId): DataProvider<OppiaLocale.DisplayLocale> {
```

Comment:
This & update don't seem to have test changes for them. I suggest adding tests to verify that the language is being saved correctly on-disk (which you can verify by recreating the test component and loading the language; you can check that the test works by temporarily reverting to the old TranslationController and seeing that the test then fails).

It's also useful to verify that the language is correctly kept per-profile, and to add these tests if they don't already exist.

Reply:
Hey Ben i have added tests to verify saving and also per profile caching. After creating another profile on the App itself just noted that Languages are kept correctly in options but not on the UI itself. If i make language change for a given profile it affects the UI language of another profile but not on the OptionsLayout and LanguageSelectionFragment. Investigating this scenario.

Reply:
Has this now been addressed @KevinGitonga?

Reply:
Updated.

Reply:
Is the first part being verified? I didn't notice any new tests verifying that the on-disk storage part is happening.

Reply:
HI ben please find here a few more covering getAppLanguageLocale
https://github.com/KevinGitonga/oppia-android/blob/60b8034f926d27a6bf9c771f8f5d76a6ec7c83f1/domain/src/test/java/org/oppia/android/domain/translation/TranslationControllerTest.kt#L338
https://github.com/KevinGitonga/oppia-android/blob/60b8034f926d27a6bf9c771f8f5d76a6ec7c83f1/domain/src/test/java/org/oppia/android/domain/translation/TranslationControllerTest.kt#L251